### ISSUE TITLE
Parameters setting in `FakeMaven` does not work on execution `AssembleMojo`

### ISSUE THREAD

**Issue by maxonfjvipon** [objectionary/eo#2406]
I'm trying to test how `AssembleMojo` works with [versions](#1602), my first attempt was:
```java
@Test
void assemblesTogetherWithVersions(@TempDir final Path temp) throws Exception {
    final Map<String, CommitHash> hashes = new CommitHashesMap.Fake();
    final CommitHash master = hashes.get("master");
    final CommitHash five = hashes.get("0.28.5");
    final Map<String, Path> result = new FakeMaven(temp)
        .withVersionedHelloWorld()
        .with("withVersions", true)
        .with(
            "objectionaries",
            new ObjsDefault(
                new MapEntry<>(master, new OyRemote(master)),
                new MapEntry<>(five, new OyRemote(five)),
            )
        )
        .execute(AssembleMojo.class)
        .result();
    MatcherAssert.assertThat(
        result.get("target/classes").toAbsolutePath(),
        new ContainsFile("**/eo-runtime-0.28.5.jar")
    );
}
```
On `resolve` step I got an error:
```
15:09:04 [ERROR] org.eolang.maven.ResolveMojo: 1 conflicting dependencies are found: {org.eolang:eo-runtime:jar:=[0.28.5, 0.29.5]}
```
Then I tried to set `ignoreVersionConflicts` flag to `true`, now my code looks like this:
```java
@Test
void assemblesTogetherWithVersions(@TempDir final Path temp) throws Exception {
    final Map<String, CommitHash> hashes = new CommitHashesMap.Fake();
    final CommitHash master = hashes.get("master");
    final CommitHash five = hashes.get("0.28.5");
    final Map<String, Path> result = new FakeMaven(temp)
        .withVersionedHelloWorld()
        .with("withVersions", true)
        .with("ignoreVersionConflicts", true) // <--- flag is set here
        .with(
            "objectionaries",
            new ObjsDefault(
                new MapEntry<>(master, new OyRemote(master)),
                new MapEntry<>(five, new OyRemote(five)),
            )
        )
        .execute(AssembleMojo.class)
        .result();
    MatcherAssert.assertThat(
        result.get("target/classes").toAbsolutePath(),
        new ContainsFile("**/eo-runtime-0.28.5.jar")
    );
}
```
I expect it works correctly and ignores conflicts but I still get the same error:
```
15:09:04 [ERROR] org.eolang.maven.ResolveMojo: 1 conflicting dependencies are found: {org.eolang:eo-runtime:jar:=[0.28.5, 0.29.5]}
```

**Comment by maxonfjvipon** [comment]
I made a debugging of `ProbeMojo` and found out that even after I did `with("ignoreVersionConflicts", true)` this flag inside `ProbeMojo` is still `false`.
`ignoreVersionConflicts` flag exists only inside `ProbeMojo`, it does not exist in `SafeMojo` or `AssembleMojo`. So the problem is that we can't set "such" flags in certain mojos via `FakeMaven` when `AssembleMojo` is executed. "Such" means "that exist only inside certain mojo but not in `AssembleMojo` or `SafeMojo`". Examples of such flags:
- "hsh" in `ProbeMojo` and `PullMojo`
- "tag" in `ProbeMojo` and `PullMojo`

**Comment by Yanich96** [comment]
@maxonfjvipon please assign me

**Comment by 0pdd** [comment]
@maxonfjvipon the puzzle [#2679](https://github.com/objectionary/eo/issues/2679) is still not solved; solved: [#2668](https://github.com/objectionary/eo/issues/2668).

**Comment by maxonfjvipon** [comment]
The main reason of such behaviour - mismatch of parameters between "parent" mojo (AssembleMojo) and "child" mojos (ResolveMojo is such case). When we put every field and parameter from child mojos to master mojo - everything works fine. #2679 - is the last relevant ticket related to the problem.

**Comment by 0pdd** [comment]
@maxonfjvipon the puzzle [#3782](https://github.com/objectionary/eo/issues/3782) is still not solved; solved: [#2668](https://github.com/objectionary/eo/issues/2668), [#2679](https://github.com/objectionary/eo/issues/2679).

**Comment by 0pdd** [comment]
@maxonfjvipon all 3 puzzles are solved here: [#2668](https://github.com/objectionary/eo/issues/2668), [#2679](https://github.com/objectionary/eo/issues/2679), [#3782](https://github.com/objectionary/eo/issues/3782).
